pkg/model/utxo/utxo.go

```diff
@@ -355,3 +357,41 @@ func (u *Manager) AddUnspentOutput(unspentOutput *Output) error {
 
 	return mutations.Commit()
 }
+
+func (u *Manager) LedgerStateSHA256Sum() ([]byte, error) {
+	u.ReadLockLedger()
+	defer u.ReadUnlockLedger()
+
+	ledgerStateHash := sha256.New()
+
+	ledgerIndex, err := u.ReadLedgerIndexWithoutLocking()
+	if err != nil {
+		return nil, err
+	}
+	if err := binary.Write(ledgerStateHash, binary.LittleEndian, ledgerIndex); err != nil {
+		return nil, err
+	}
+
+	// get all UTXOs and sort them by outputID
+	var outputs LexicalOrderedOutputs
+	if err := u.ForEachUnspentOutput(func(output *Output) bool {
+		outputs = append(outputs, output)
```

Comment:
this reads all outputs into memory. With stardust this is probably a bad idea.

Reply:
I will fix this in another PR :+1: